Source/Applications/SystemCenter/wwwroot/Scripts/TSX/SystemCenter/Settings/DataReaders.tsx

```diff
@@ -76,97 +62,108 @@ const DataReaders: Application.Types.iByComponent = (props) => {
         setErrors(e)
     }, [editnewSetting])
 
-    const searchFields: Search.IField<OpenXDA.Types.DataReader>[] = [
-        { key: 'FilePattern', label: 'File Pattern', type: 'string', isPivotField: false },
-        { key: 'AssemblyName', label: 'Assembly Name', type: 'string', isPivotField: false },
-        { key: 'TypeName', label: 'Type Name', type: 'string', isPivotField: false },
-        { key: 'LoadOrder', label: 'Load Order', type: 'number', isPivotField: false }
-    ]
-
-    if (status === 'error')
-        return <div style={{ width: '100%', height: '100%' }}>
-            <ServerErrorIcon Show={true} Label={'A Server Error Occurred. Please Reload the Application.'} />
-        </div>;
+    function addOperation() {
```

Comment:
These calls look like standard model controller calls, can we use the controller object here like in other places?

Reply:
updated